controllers/storagecluster/generate.go

```diff
@@ -83,3 +91,38 @@ func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string)
 func generateStorageQuotaName(storageClassName, quotaName string) string {
 	return fmt.Sprintf("ocs-%s-%s", storageClassName, quotaName)
 }
+
+// generateCephFSProviderParameters function generates extra parameters required for provider storage clusters
+func generateCephFSProviderParameters(initData *ocsv1.StorageCluster) map[string]string {
+	if initData.Spec.AllowRemoteStorageConsumers {
+		deviceSetList := initData.Spec.StorageDeviceSets
+		var deviceSet *ocsv1.StorageDeviceSet = nil
+		for _, ds := range deviceSetList {
+			if ds.Name == StorageDeviceName {
+				deviceSet = &ds
+			}
+		}
```

Comment:
Unless `deviceSetList` is an array of pointers (which I don't think it is), this form of `for-range` loop will create a in-memory copy of the device set object on each iteration, and changes to the resulting object will not be reflected in the original object (the one in the list). 
Please use indexing into the original array to get the address of the source device set 

```suggestion 
		for i := range deviceSetList {
			ds := &deviceSetList[i] 
			if ds.Name == StorageDeviceName {
				deviceSet = ds
			}
		}
```